protocol/p2p/src/core/router.rs

```diff
@@ -269,6 +317,7 @@ impl Router {
     /// This should be used by `ConnectionInitializer` instances to register application-specific flows.
     pub fn subscribe_with_capacity(&self, msg_types: Vec<KaspadMessagePayloadType>, capacity: usize) -> IncomingRoute {
         let (sender, receiver) = mpsc_channel(capacity);
+        let incoming_route = IncomingRoute::new(receiver);
```

Comment:
technically we can maintain a route id counter per router instead of a global static one (no problem with duplicate ids cross different connections)